src/net.cpp

```diff
@@ -590,25 +599,35 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes, bool& complete
     while (nBytes > 0) {
         // absorb network data
         int handled = m_deserializer->Read(pch, nBytes);
-        if (handled < 0) return false;
+        if (handled < 0) {
+            // Serious header problem, disconnect from the peer.
+            return false;
+        }
 
         pch += handled;
         nBytes -= handled;
 
         if (m_deserializer->Complete()) {
             // decompose a transport agnostic CNetMessage from the deserializer
-            CNetMessage msg = m_deserializer->GetMessage(Params().MessageStart(), time);
+            uint32_t out_err{0};
+            Optional<CNetMessage> result{m_deserializer->GetMessage(time, out_err)};
+            if (!result) {
+                // Message deserialization failed.  Drop the message but don't disconnect the peer.
+                // store the size of the corrupt message
+                mapRecvBytesPerMsgCmd.find(NET_MESSAGE_COMMAND_OTHER)->second += out_err;
```

Comment:
I suggest you use a new category NET_MESSAGE_COMMAND_CORRUPT here rather than overloading _OTHER.

Reply:
I'm worried about how adding a new category will affect people who have scripts that parse `getpeerinfo`.  They will either:

1. Crash (bad)
2. Ignore it (also bad, loses the "sum of messages in bytesrecv_per_msg" == "bytesrecv" invariant)
3. Treat it like "_OTHER" (same as what we have here)

As it is right now in the PR, the bytes received for corrupt messages have been moved from their respective command to _OTHER.  This is definitely an accuracy improvement.  I'm not sure that adding a new category is worth the scope creep.

I've implemented your suggested change [at this branch](https://github.com/troygiorshev/bitcoin/tree/2020-07-p2p-corrupt).

Reply:
We _do_ occasionally add new message types (most recently in #18876), so a client should tolerate unrecognized message types (crashing/requiring a software update for the client every time we introduce a new message type would be bad).

I agree that this isn't a critical issue and could be changed in a follow-up if necessary. For now, _OTHER seems fine.